examples/rust/service_metrics/src/bin/client.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures::StreamExt;
+use service_metrics::DEFAULT_NAMESPACE;
+
+// use tokio::time::{Duration, Instant};
+use triton_distributed::{
+    logging,
+    protocols::annotated::Annotated,
+    utils::{stream, Duration, Instant},
+    DistributedRuntime, Result, Runtime, Worker,
+};
+
+fn main() -> Result<()> {
+    logging::init();
+    let worker = Worker::from_settings()?;
+    worker.execute(app)
+}
+
+async fn app(runtime: Runtime) -> Result<()> {
+    let distributed = DistributedRuntime::from_settings(runtime.clone()).await?;
+
+    let namespace = distributed.namespace(DEFAULT_NAMESPACE)?;
+    let component = namespace.component("backend")?;
+
+    let client = component
+        .endpoint("generate")
+        .client::<String, Annotated<String>>()
+        .await?;
+
+    client.wait_for_endpoints().await?;
+
+    let mut stream = client.random("hello world".to_string().into()).await?;
+
+    while let Some(resp) = stream.next().await {
+        println!("{:?}", resp);
+    }
+
+    let metrics = distributed
+        .nats_client()
+        .scrape_service(&component.service_name())
+        .await?;
```

Comment:
Calling scrape_service on the nats_client seems a little awkward to me for user-facing code. Can this be simplified to something like:
```suggestion
    let metrics = distributed
        .scrape_service(&component.service_name())
        .await?;
```

and internally on the DistributedRuntime method it hides/wraps the nats_client part?
```rust
self.nats_client().scrape_service(...)
```

Reply:
This also looks a little similar to `component.scrape_stats()` which calls `drt.sevice_client().collect_services()` which calls `self.nats_client().scrape_service()` - maybe we can simplify some things here

Reply:
yes, we have to formalize this across the stack. lots to do here.

Reply:
done